js/fairdesk.js

```diff
@@ -0,0 +1,1450 @@
+            'symbol': market['id'],
+        };
+        const response = await this.publicGetMdTicker24h (this.extend (request, params));
+        // {
+        //     "status": 0,
+        //     "error": "OK",
+        //     "data": {
+        //       "symbol": "btcusdt",
+        //       "timestamp": 1651895571086,
+        //       "open": 36439.500000,
+        //       "high": 36542.500000,
+        //       "low": 35344.500000,
+        //       "close": 35890.000000,
+        //       "indexPrice": 35891.0,
+        //       "markPrice": 35892.9,
+        //       "openInterest": 74.275366,
+        //       "fundingRate": 0.00010000,
+        //       "predicateFundingRate": 0.000100,
+        //       "baseVolume": 313.214000,
+        //       "quoteVolume": 11280153.148
+        //     }
+        // }
+        const result = this.safeValue (response, 'data', {});
+        return this.parseTicker (result, market);
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        const symbolId = this.parseSymbol (symbol);
+        await this.loadMarkets ();
+        const market = this.market (symbolId);
+        const request = {
+            'symbol': market['id'],
+        };
+        let method = 'publicGetMdTradeRecent';
+        if (since) {
+            method = 'publicGetMdTradeHistory';
+            request['from'] = since;
+        }
+        const response = await this[method] (this.extend (request, params));
+        // {
+        //     "status": 0,
+        //     "error": "OK",
+        //     "data": {
+        //       "symbol": "btcusdt",
+        //       "timestamp": 1651895759240,
+        //       "trades": [{
+        //         "tradeId": 245418671,
+        //         "orderId": 45014798997,
+        //         "qty": 38408.500000,
+        //         "price": 38408.500000,
+        //         "timestamp": 1651447203805,
+        //         "updateId": 47240871986,
+        //         "taker": true,
+        //         "buyMaker": false
+        //       },]
+        //     }
+        // }
+        const result = this.safeValue (response, 'data', {});
+        const trades = this.safeValue (result, 'trades', []);
+        return this.parsePubTrades (trades, market, since, limit);
+    }
+
+    parsePubTrades (trades, market = undefined, since = undefined, limit = undefined, params = {}) {
+        const result = [];
+        for (let i = 0; i < trades.length; i++) {
+            const item = trades[i];
+            result.push (this.parsePubTrade (item, market));
+        }
+        return result;
+    }
+
+    parseMyTrades (trades, market = undefined, since = undefined, limit = undefined, params = {}) {
+        const result = [];
+        for (let i = 0; i < trades.length; i++) {
+            const item = trades[i];
+            result.push (this.parseMyTrade (item, market));
+        }
+        return result;
+    }
+
+    parsePubTrade (trade, market = undefined) {
```

Comment:
We're using the same parser for both the public and the private trades, see how parseTrade is implemented based on strings with safeTrade in the other exchanges.